Sources/KSCrashRecordingCore/include/KSCrashMonitor.h

```diff
@@ -43,40 +43,63 @@ extern "C" {
 
 struct KSCrash_MonitorContext;
 
+typedef struct
+{
+    const char* (*monitorId)(void);
+    KSCrashMonitorFlag (*monitorFlags)(void);
+    void (*setEnabled)(bool isEnabled);
+    bool (*isEnabled)(void);
+    void (*addContextualInfoToEvent)(struct KSCrash_MonitorContext* eventContext);
+    void (*notifyPostSystemEnable)(void);
+} KSCrashMonitorAPI;
 
 // ============================================================================
 #pragma mark - External API -
 // ============================================================================
 
-/** Set which monitors are active.
+/**
+ * Activates all added crash monitors.
+ */
+void kscm_activateMonitors(void);
+
+/**
+ * Disables all active crash monitors.
+ */
+void kscm_disableAllMonitors(void);
+
+/**
+ * Adds a crash monitor to the system.
  *
- * @param monitorTypes Which monitors should be active.
+ * @param api Pointer to the monitor's API.
  */
-void kscm_setActiveMonitors(KSCrashMonitorType monitorTypes);
+void kscm_addMonitor(KSCrashMonitorAPI* api);
 
-/** Get the currently active monitors.
+/**
+ * Remove a crash monitor from the system.
+ *
+ * @param api Pointer to the monitor's API.
  */
-KSCrashMonitorType kscm_getActiveMonitors(void);
+void kscm_removeMonitor(KSCrashMonitorAPI* api);
```

Comment:
nit: Would be also good to add `const` ti these methods. Not a requirement, but good.